hal/armv7m/imxrt/timer.c

```diff
@@ -75,14 +115,38 @@ int hal_timerRegister(int (*f)(unsigned int, cpu_context_t *, void *), void *dat
 
 void _hal_timerInit(u32 interval)
 {
-	timer_common.jiffies = 0;
-
-	_imxrt_systickInit(interval);
-
 	timer_common.interval = interval;
+	timer_common.upper = 0;
 	hal_spinlockCreate(&timer_common.sp, "timer");
 	timer_common.handler.f = _timer_irqHandler;
-	timer_common.handler.n = SYSTICK_IRQ;
+	timer_common.handler.n = GPT_IRQ;
 	timer_common.handler.data = NULL;
 	hal_interruptsSetHandler(&timer_common.handler);
+
+	timer_common.base = (void *)GPT_BASE;
+
+	/* Disable timer */
+	*(timer_common.base + gpt_cr) &= ~1;
```

Comment:
Probably should be the first thing we do - GPT1 interrupts are used by PLO.

Reply:
plo is bound to deinit all hw it's using before jump to kernel (even including uarts), but it doesn't hurt to be safe, so changed